src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1549,15 +1584,13 @@ uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uct_ib_mem_t *ib_memh)
 err_cross_mr_destroy:
     mlx5dv_devx_obj_destroy(memh->cross_mr);
     memh->cross_mr = NULL;
-err_umem_dereg:
-    mlx5dv_devx_umem_dereg(memh->umem);
-    memh->umem = NULL;
 err_out:
-    memh->super.exported_lkey = UCT_IB_INVALID_MKEY;
+    if (memh->umem != NULL) {
```

Comment:
do we also need to set `memh->super.exported_lkey = UCT_IB_INVALID_MKEY;`?
E.g. its calue is checked in `uct_ib_mkey_pack`